config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,46 @@
+class Rack::Attack
+  class Request < ::Rack::Request
+    # Take remote IP from Cloudfare's headers instead of rev proxy IP
+    def remote_ip
+      # Cloudflare stores remote IP in CF_CONNECTING_IP header
+      @remote_ip ||= (env['HTTP_CF_CONNECTING_IP'] ||
+                      env['action_dispatch.remote_ip'] ||
+                      ip).to_s
+    end
+  end
+  # Disable if DISABLE_RACK_ATTACK and if env is not production
+  if !Rails.env.production?
+    Rack::Attack.enabled = !ENV['DISABLE_RACK_ATTACK']
+  end
+
+  # Use redis for rack attack since config.cache_store is not set
+  redis_client = Redis.new
+  Rack::Attack.cache.store = Rack::Attack::StoreProxy::RedisStoreProxy.new(redis_client)
+
+  # Throttle all non asset requests
+  throttle('req/ip', limit: 300, period: 5.minutes) do |req|
```

Comment:
This is not needed as we are already under cloudflare, also this can be tricky when multiple students accessing circuitverse from college lab as external IP will be same for all